File: Daria-Goldenberg/SecondTerm/Task 2/BlackJack/Hand.cs
﻿namespace BlackJack
{
	public class Hand
	{
		public List<Card> Cards { get; private set; }

		public Hand()
		{
			Cards = new List<Card>();
		}

		public void TakeCard(Deck deck)
		{
			Cards.Add(deck.TakeCard());
		}

		public void Clear()
		{
			Cards.Clear();
		}

		public int CountPoints()
		{
			int score = 0;
			foreach (Card card in Cards)
				score += card.GetValue(score);

			return score;
		}
	}
}

File: Vadim-Yakshigulov/Semester 2/Blackjack/src/main/kotlin/blackjack/classes/Hand.kt
package blackjack.classes

import blackjack.interfaces.IBetPlaced
import blackjack.interfaces.ICard
import blackjack.interfaces.IHand

class Hand : IHand, IBetPlaced {
    private val _cards: MutableList<ICard> = mutableListOf()
    override val cards: List<ICard>
        get() = _cards
    override var blocked: Boolean = false

    override var bet: Int = 0

    override fun dealCard(): ICard {
        return _cards.removeLast()
    }

    override fun splitByLastCard(): Hand {
        val lastCard = _cards.removeLast()
        return Hand().apply { addCard(lastCard) }
    }

    override fun addCards(vararg cards: ICard) {
        cards.forEach { c -> this.addCard(c) }
    }

    override fun addCard(card: ICard) {
        _cards.add(card)
    }

    override fun flipAllCards(isFaceUp: Boolean) {
        for (card in _cards)
            card.isFaceUp = isFaceUp
    }

    override fun clear() {
        _cards.clear()
    }
}